src/vss_tools/vspec/vssexporters/vss2samm/helpers/vssHelper.py

```diff
@@ -0,0 +1,569 @@
+            return get_parent_prefix_for_ttl_name(vss_node.parent, ttl_name, use_vehicle_prefix=True)
+
+    return parent_prefix
+
+
+# Set ttl_name for provided vss_node.
+# BY DEFAULT: ttl_name will be set to the current vss_node.name.
+#   vss_node           - the VSSNode to be updated
+#   use_parent_prefix  - when provided, the vss_node.parent.name will be used as prefix to its vss_node.name,
+#                        if the vss_node has a parent
+#   overwrite_ttl_name - if specified and the provided vss_node, already has a ttl_name,
+#                        it will be overwritten based on preserve_ttl_name
+#   preserve_ttl_name  - if specified and the provided vss_node, already has a ttl_name,
+#                        then its ttl_name will be preserved and will be prefixed with its parent name
+def set_ttl_name(vss_node: VSSNode, use_parent_prefix: bool, overwrite_ttl_name=False, preserve_ttl_name=False):
+
+    if vss_node.ttl_name and not overwrite_ttl_name:
+        log.debug(
+            "Node ttl name: '%s' was already set.",
+            vss_node.ttl_name
+        )
+
+    else:
+        ttl_name = vss_node.ttl_name if vss_node.ttl_name and preserve_ttl_name else vss_node.name
+
+        if use_parent_prefix:
+            parent_prefix = get_parent_prefix_for_ttl_name(vss_node, ttl_name)
+            vss_node.ttl_name = parent_prefix + ttl_name
+
+        else:
+            vss_node.ttl_name = ttl_name
+
+        log.debug(
+            "%s node ttl name: '%s'.",
+            "updated" if overwrite_ttl_name else "added",
+            vss_node.ttl_name
+        )
+
+
+# Helper function to clone a provided VSSNode, instead of to copy it, using the python available copy functionality,
+# which will copy the whole node.
+#
+# This is required so we can have a "clean" unbound to properties of the provided VSSNode, copy of it,
+# which then can be further manipulated.
+#
+# For example: filter the children of cloned VSSNode and add them to its cloned (copied) version.
+# REASON FOR THIS is that:
+#       VSSNode.children is immutable, when the node had some children
+#       BUT: we can set (add) them, in case when VSSNode.children was initially empty.
+def clone_vss_node(vss_node: VSSNode):
+    # NOTE: Below properties names are taken from: VSSNode: validate_vss_element function.
+    #       If required to sync these, use the aforementioned function as reference.
+    #
+    #       Currently we skip the "children" one,
+    #       as these will be further filtered and populated in the cloned_vss_node.
+    vss_node_dict = {}
+
+    # key:value pairs where:
+    #     key   - is the VSSNode property
+    #     value - is the VSS specification, property,
+    #             which was used for the creation of vss_node,
+    #             i.e. its corresponding name in the used for creation of VSSNode, dictionary.
+    properties_to_clone = {
+        "$file_name$"   : "$file_name$",    # noqa: E203
```

Comment:
Might be simpler to specify things not to clone? But yeah, see comment before

Reply:
As per my previous comment I was trying to keep only the fields which my script was going to use, instead of copying whole VssNode, but again as I said I am OK for further refactoring of this and if possible to build "filtered" tree in different way I can even remove it.

Reply:
The "cloning" and related logic is now removed.

Done